### PR TITLE
FIX Issue 19643 - add missing Objective-C function calling convention

### PR DESCRIPTION
**Pull request by kubo39**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
LGTM, cc @jacob-carlborg

File: src/core/demangle.d [hunk 1]
```diff
@@ -2389,21 +2403,24 @@ private template hasPlainMangling(FT) if (is(FT == function))
     static extern(C) void fooC();
     static extern(Windows) void fooW();
     static extern(C++) void fooCPP();
+    static extern(Objective-C) void fooObjC();
```
Comment (jacob-carlborg) [inline src/core/demangle.d]: Not sure if this works, but `extern(Objective-C)` only makes sense on classes.

**Comment by rainers** [comment]
The mangling for Objective-C has an ambiguity if used with the grammar from the website, see https://issues.dlang.org/show_bug.cgi?id=17310. IIRC the demangler should be able to handle these nevertheless, but you might want to add some symbols from that bug report to the tests, too.

File: src/core/demangle.d (hunk 1 above)
Comment (kubo39) [reply src/core/demangle.d]: I think `extern (Objective-C)` with classes works only if `macOS && 64bit`.
Should I use `version(OSX) { extern(Objective-C) class Foo { ... ` ?

Comment (jacob-carlborg) [reply src/core/demangle.d]: Yes, but use `version(D_ObjectiveC)` instead.

**Pull request closed without merging**